pvacseq/lib/binding_filter.py

```diff
@@ -32,140 +30,65 @@ def main(args_input = sys.argv[1:]):
-        #7	transcript_name
-        #8	amino_acid_change
-        #9	ensembl_gene_id
-        #10	wildtype_amino_acid_sequence
-
-        gene = data[5]
-        amino_acid_change = data[7]
-        variants[ gene + '\t' + amino_acid_change] = [word for word in data]
-
-    #dump header data to output file
-    args.input.close()
-    if 'header' not in variants:
-        # usage("Header not defined in variant input file")
-        parser.print_help()
-        print("\nError: Header not defined in variant input file")
-        exit(1)
-    output = csv.writer(args.output, delimiter='\t', lineterminator='\n')
-
-    output.writerow([
-        *variants['header'],
-        "GeneName",
-        "HLAallele",
-        "PeptideLength",
-        "SubPeptidePosition",
-        "MTScore",
-        "WTScore",
-        "MTEpitopeSeq",
-        "WTEpitopeSeq",
-        "FoldChange"
-    ])
-
-    #Read netmhc files from the fof, and parse into predictions
     prediction = {}
-
-    intake = args.fof.readline().rstrip()
-
-
-    while intake != "":
-
-        #parse the filepath
-        basename = os.path.basename(intake)
-        data = basename.split(".")
-        sample = netmhc_subber.sub("", data[0])
-        allele = data[1]
-        length = data[2]
-        mode = 'filtered'
-
-        if sample not in prediction:
-            prediction[sample] = {}
-        #open each file listed in the fof, and read gene data
-        netmhc_file = open(intake, mode = 'r')
-        netmhc_file.readline() #skip first line
-        netmhc_reader = csv.reader(netmhc_file, delimiter='\t')
-        for line_data in netmhc_reader:
-
-            gene = {
-                'gene_name' : line_data[0],
-                'allele' : allele,
-                'mode' : mode,
-                'length' : length,
-                'point_mutation' : line_data[1],
-                'sub_peptide_position' : line_data[2],
-                'mt_score' : line_data[3],
-                'wt_score' : line_data[4],
-                'mt_epitope_seq' : line_data[5],
-                'wt_epitope_seq' : line_data[6],
-                'fold_change' : line_data[7]
+    fieldnames = []
+
+    for line in args.fof:
+        filepath = line.rstrip()
+        sample = filepath.split(".")[0].replace("_netmhc", "")
+        if not os.path.isfile(filepath):
+            #if path does not exist, try making it relative to the fof
+            filepath = os.path.join(
+                os.path.dirname(os.path.abspath(args.fof.name)),
+                filepath
+            )
+        file_handle = open(filepath, mode='r')
+        reader = csv.DictReader(file_handle, delimiter='\t')
+        if len(fieldnames) == 0:
+            fieldnames = reader.fieldnames
+
+        prediction[sample] = {}
+
+        for entry in reader:
+            name = entry['Gene Name']
+            score = int(entry['MT score'])
+            fold_change = 0 if entry['Fold Change'] == 'NA' else float(entry['Fold Change'])
```

Comment:
The latter  - Filter only on binding threshold. Essentially, they should be treated as having "Inf" in their fold-change column, since they have infinitely better binding than the non-existent wildtype